File: JULY-2025/01-07-2025/p3330.scala
object Solution {
    def possibleStringCount(word: String): Int = {
        word.toCharArray.foldLeft((0, word.head)) {
            case ((count, prev), value) =>
                if (prev == value) (count + 1, value) else (count, value)
        }._1
    }
}

File: JULY-2025/01-07-2025/p3330.dart
class Solution {
  int possibleStringCount(String word) {
    if (word.isEmpty) return 0;
    int count = 1;
    int currentLength = 1;
    for (int i = 1; i < word.length; i++) {
        if (word[i] == word[i - 1]) {
            currentLength++;
        } else {
            count += currentLength - 1;
            currentLength = 1;
        }
    }
    count += currentLength - 1;
    return count;
  }
}